### ISSUE TITLE
`show math.equation: set align(...)` breaks alignment points

### ISSUE THREAD

**Issue by EpicEricEE** [typst/typst#3973]
### Description

The alignment set with `show math.equation: set align(...)` overrides the left-right alternating behavior of alignment points:

```typ
$ a + b &= c \
      e &= f + g + h $
```

| Alignment | Output |
| :---: | :---: |
| _default_ |  ![image](https://github.com/typst/typst/assets/7191192/cd6ad2d3-12b0-46e9-8d9b-5f53d4c9450e) |
| `start` | ![image](https://github.com/typst/typst/assets/7191192/763ecb6c-f29f-453c-8cd0-d7074e23b3cc) |
| `end` | ![image](https://github.com/typst/typst/assets/7191192/307386f0-4c03-41f2-8402-fed4a8fa8ad0) |

### Operating system

Web app

### Typst version

- [X] I am using the latest version of Typst

**Comment by Leedehai** [comment]
I'm not sure this is a bug.. 🤔  because the code below seems intentional to only use `alternator` in the `_`-case:
https://github.com/typst/typst/blob/0bb45b335fb8b34af9ce33e1684e58c046ee5e45/crates/typst/src/math/row.rs#L226-L240
@laurmaedje thoughts?

**Comment by SUPERCILEX** [comment]
Yeah, that's how it's supposed to work. If OP's goal to align the whole equation in its outer context, does putting it in a box and aligning that work instead?

**Comment by EpicEricEE** [comment]
This show-set rule is the intended way of setting the "main" alignment of block equations (see #2156, #2157 and [on Discord](https://discord.com/channels/1054443721975922748/1054443722592497796/1152334279745994863)), so I would argue that it's not intended.

**Comment by SUPERCILEX** [comment]
I guess does anybody use the feature? We could get rid of it, but I vaguely recall some discussion that led to wanting to be able to change how internal equation alignment worked. For OP's purposes, using left/right instead should get the job done now that I'm rereading this.

**Comment by laurmaedje** [comment]
I agree that the current behaviour is unexpected. We should probably just get rid of it. 

> I vaguely recall some discussion that led to wanting to be able to change how internal equation alignment worked.

Do you happen to be able to still find that?

**Comment by SUPERCILEX** [comment]
Nope, so feel free to get rid of it. :)

**Comment by PgBiel** [comment]
I think the main problem here is that alignment in math equations and text alignment are currently treated as two very separate things, so this behavior comes off as surprising. It seems perhaps ideal to remove this for the time being and maybe add some other option to `math.equation` itself in the future, if needed.
